File: flutter/lib/domain/entities/campus.dart
import 'package:flutter/foundation.dart';

/// Entity representando tabela `campus` (ap_achados_perdidos.campus)
/// Corresponde a: com.AchadosPerdidos.API.Domain.Entity.Campus
@immutable
class Campus {
  final int id;
  final String nome;
  final int instituicaoId;
  final int enderecoId;
  final DateTime? dtaCriacao;
  final bool flgInativo;
  final DateTime? dtaRemocao;

  const Campus({
    required this.id,
    required this.nome,
    required this.instituicaoId,
    required this.enderecoId,
    this.dtaCriacao,
    this.flgInativo = false,
    this.dtaRemocao,
  });
}

File: api/src/main/java/com/AchadosPerdidos/API/Domain/Entity/Campus.java
package com.AchadosPerdidos.API.Domain.Entity;

import java.util.Date;

public class Campus {
    private Integer id;
    private String nome;
    private Integer instituicaoId;
    private Integer enderecoId;
    private Date dtaCriacao;
    private Boolean flgInativo;
    private Date dtaRemocao;

    public Campus() {}

    public Campus(Integer id, String nome, Integer instituicaoId, Integer enderecoId, Date dtaCriacao, Boolean flgInativo, Date dtaRemocao) {
        this.id = id;
        this.nome = nome;
        this.instituicaoId = instituicaoId;
        this.enderecoId = enderecoId;
        this.dtaCriacao = dtaCriacao;
        this.flgInativo = flgInativo;
        this.dtaRemocao = dtaRemocao;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Integer getInstituicaoId() {
        return instituicaoId;
    }

    public void setInstituicaoId(Integer instituicaoId) {
        this.instituicaoId = instituicaoId;
    }

    public Integer getEnderecoId() {
        return enderecoId;
    }

    public void setEnderecoId(Integer enderecoId) {
        this.enderecoId = enderecoId;
    }

    public Date getDtaCriacao() {
        return dtaCriacao;
    }

    public void setDtaCriacao(Date dtaCriacao) {
        this.dtaCriacao = dtaCriacao;
    }

    public Boolean getFlgInativo() {
        return flgInativo;
    }

    public void setFlgInativo(Boolean flgInativo) {
        this.flgInativo = flgInativo;
    }

    public Date getDtaRemocao() {
        return dtaRemocao;
    }

    public void setDtaRemocao(Date dtaRemocao) {
        this.dtaRemocao = dtaRemocao;
    }

}

